From 1f5fdbb40fc22731bb6a5454f7f1710cf808f01c Mon Sep 17 00:00:00 2001
From: Vincent Abriou <vincent.abriou@st.com>
Date: Mon, 23 Jan 2017 17:23:39 +0100
Subject: [PATCH 4/4] compositor: force repaint immediately on receiving the
 pageflip completion event

This is a temporary patch that allow to reach the maximum display
refresh framerate when the cpu frequency scaling is activated.

Indeed, weston relies on the Linux hrtimer to schedule the repaint
processing but with the frequency scaling activated, the hrtimer does
not fired at the expected time.
So, by forcing repaint immediately on receiving the pageflip completion
event, this maximizes the time available for the compositor itself to
repaint, but it also means that clients can never hit the very next
vblank / pageflip.

For more information on the repaint scheduling algorithm please look at
this web page:
https://www.collabora.com/about-us/blog/2015/02/12/weston-repaint-scheduling/

Note: Do not upstream this patch

Change-Id: I61ad46c9ce5dbd3e7daed1cc39cbe8578f40fb06
Signed-off-by: Vincent Abriou <vincent.abriou@st.com>
Reviewed-on: https://gerrit.st.com/63903
Reviewed-by: Fabien DESSENNE <fabien.dessenne@st.com>
Tested-by: Fabien DESSENNE <fabien.dessenne@st.com>
---
 libweston/compositor.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libweston/compositor.c b/libweston/compositor.c
index 2057b0d..9e7c34e 100644
--- a/libweston/compositor.c
+++ b/libweston/compositor.c
@@ -64,7 +64,9 @@
 #include "version.h"
 #include "plugin-registry.h"
 
-#define DEFAULT_REPAINT_WINDOW 7 /* milliseconds */
+/* Force repaint scheduling algorithm to repaint immediatly on receiving the
+ * pageflip completion event */
+#define DEFAULT_REPAINT_WINDOW 100 /* milliseconds */
 
 static void
 weston_output_transform_scale_init(struct weston_output *output,
-- 
2.7.4

